mam/examples/ntru-key-exchange.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2019 IOTA Stiftung
+ * https:github.com/iotaledger/entangled
+ *
+ * MAM is based on an original implementation & specification by apmi.bsu.by
+ * [ITSec Lab]
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include <stdio.h>
+#include "mam/api/api.h"
+#include "mam/ntru/ntru.h"
+
+#define SENDERSEED "SENDERSEED9SENDERSEED9SENDERSEED9SENDERSEED9SENDERSEED9SENDERSEED9SENDERSEED99999"
+#define RECEIVERSEED "RECEIVERSEED9RECEIVERSEED9RECEIVERSEED9RECEIVERSEED9RECEIVERSEED9RECEIVERSEED9999"
+#define NONCE "ABCDEFGHIJKLMNOPQRSTUVWXYZABCD"
+#define PL "PAYLOADTEST"
+
+int main() {
+  retcode_t ret = RC_OK;
+  //Setting up two APIs
+  mam_api_t sender_api;
+  tryte_t *sender_seed = (tryte_t *)SENDERSEED;
+  ERR_BIND_RETURN(mam_api_init(&sender_api, sender_seed), ret);
+  mam_api_t receiver_api;
+  tryte_t *receiver_seed = (tryte_t *)RECEIVERSEED;
+  ERR_BIND_RETURN(mam_api_init(&receiver_api, receiver_seed), ret);
+  //Generating a keypair for the receiver api
+  mam_ntru_sk_t ntru;
+  MAM_TRITS_DEF(nonce, 3 * 10);
+  nonce = MAM_TRITS_INIT(nonce, 3 * 10);
+  trits_from_str(nonce,NONCE);
+  ntru_sk_reset(&ntru);
+  ntru_sk_gen(&ntru, &receiver_api.prng, nonce);
+  //Setting the generated keypair to the receiver API
+  ERR_BIND_RETURN(mam_api_add_ntru_sk(&receiver_api, &ntru), ret);
+  //Adding the receiver APIs public key to the sender API public key list
+  ERR_BIND_RETURN(mam_api_add_ntru_pk(&sender_api, &ntru.public_key), ret);
+  //Preparing channel and endpoint with sender API with randomly chosen height
+  tryte_t channel_id[MAM_CHANNEL_ID_TRYTE_SIZE];
+  ERR_BIND_RETURN(mam_api_channel_create(&sender_api, 2, channel_id), ret);
+  tryte_t endpoint_id[MAM_ENDPOINT_ID_TRYTE_SIZE];
+  ERR_BIND_RETURN(mam_api_endpoint_create(&sender_api, 2, channel_id, endpoint_id), ret);
+  //prepare bundle for sender API to write to
+  bundle_transactions_t *bundle = NULL;
+  bundle_transactions_new(&bundle);
+  //prepare and write header and packet
+  trit_t message_id[MAM_MSG_ID_SIZE];
+  ERR_BIND_RETURN(mam_api_bundle_write_header_on_endpoint(&sender_api, channel_id, endpoint_id, NULL, NULL, bundle, message_id), ret);
```

Comment:
Do I really not need to specify the ntru public keys here?

Reply:
For now at least you will have too because the keys inside the api struct are not used, this might change

Reply:
This is done now.